distribution/lib/Standard/Table/0.0.0-dev/src/Data/Table.enso

```diff
@@ -492,44 +491,18 @@ type Table
     aggregate columns (on_problems=Report_Warning) =
         validated = Aggregate_Column_Helper.prepare_aggregate_columns columns this
 
-        make_key = if (validated.key_columns.length == 0) then _->(Group_By_Key.key [1]) else
-            i->(Group_By_Key.key (validated.key_columns.map v->(v.at i)))
-
-        new_table = validated.valid_columns.map c->[c.first, Vector.new_builder]
-        aggregators = validated.valid_columns.map c->(Aggregate_Column_Aggregator.new c.second)
-        add_row _ =
-            idx = new_table.at 0 . at 1 . length
-            0.up_to (aggregators.length) . each i->
-                new_table.at i . at 1 . append ((aggregators.at i).initial)
-            idx
-
-        # Fold
-        aggregate map i =
-            key = make_key i
-            row_index = map.get_or_else key (add_row Nothing)
-
-            # Accumulate
-            0.up_to (validated.valid_columns.length) . each j->
-                accumulator = aggregators.at j . accumulator
-                array = new_table.at j . at 1 . to_array
-                current = array . at row_index
-                new = accumulator current i
-                array . set_at row_index new
-
-            map.insert key row_index
-
         on_problems.attach_problems_before validated.problems <|
-            # Build Table
-            if ((validated.key_columns.length == 0) && (this.row_count == 0)) then (add_row Nothing) else
-                0.up_to this.row_count . fold Map.empty aggregate
+            java_columns = validated.key_columns.map .java_column
```

Comment:
Would be good if the name made it clear that these are key columns. Maybe `java_key_columns`?

Just `java_columns` sounds like any set of columns - all columns from this table? Or something. I know it is a bit of a nitpick, but I think this helps readability quite a lot.